### PR TITLE
Upgrades packages + introduces simple integration tests

### PR DESCRIPTION
**Pull request by mattoberle**
This pull request was initially intended to update the `amphora-storage-postgres` dependencies to address security vulnerabilities.

Unfortunately, there wasn't really a good way to know if these upgrades introduced any breaking behavior to the Postgres or Redis clients (because those clients are entirely mocked during tests).

I created a new directory `tests/` with a simple integration test workflow.
Running `./tests/integration.sh` locally will create a local redis/postgres container, run the integration tests, and then destroy the containers.

The tests also run on CircleCI.

I'm hoping this can catch issues where we are utilizing a client library incorrectly, initializing settings incorrectly, etc...

### CODE REVIEW

File: tests/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        testcase.published ? '/_pages/foo/c@published' : '/_pages/foo/c',
+        testcase.published ? '/_uris/foo@published' : '/_uris/foo'
+      ],
+      vals = [
+        { test: true },
+        { test: false },
+        { test: [1 ,2 ,3] },
+        'example.com/foo'
+      ],
+      batch = keys.map((k, i) => ({ key: k, value: vals[i] }));
```
Comment (james-owen) [inline tests/integration.test.js]:
```suggestion
      batch = keys.map((key, idx) => ({ key, value: vals[idx] }));
```

File: postgres/index.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ function setup(testPostgresHost) {
     })
     .then(() => createTables())
     .then(() => ({ server: `${postgresHost}:${POSTGRES_PORT}` }))
-    .catch(logGenericError);
+    .catch(logGenericError(__filename));
```
Comment (mattoberle) [inline postgres/index.js]: `logGenericError` is a decorator function (ie. It's a function that returns a new function).

Calling it here with `__filename` results in something equivalent to:
```
.catch((err) => log('error', err,message);
```

File: tests/integration.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    await db.setup(testcase.cache);
+    await expect(db.batch(batch, testcase.cache)).resolves.toEqual(vals);
+
+    for (let i = 0; i < batch.length; i++) {
+      let key = batch[i].key, val = batch[i].value,
```
Comment (james-owen) [inline tests/integration.test.js]: Would something like this work? 
```
const batch = [ [key0, val0], [key1, val1]];

batch.forEach(([key, val]) => {
 // expect(key) => val
})
```

Comment (mattoberle) [reply tests/integration.test.js]: Oh yeah, a `.forEach` on the batch array would make sense.

Comment (mattoberle) [reply tests/integration.test.js]: Although, that is creating another function, which would also have to be async.

**Pull request merged**